driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+/**
+ * Expresses a document value. A document is an ordered set of fields, where the
+ * key is a string value, mapping to a value of any other expression type.
+ */
+public interface DocumentExpression extends Expression {
```

Comment:
This needs to clarify that this is intended to represent a record (where any particular record (class, in Java) would have a fixed set of field keys). To represent non-fixed keys, use maps (presently called MapExpressions).

Reply:
Should probably be renamed to "Record".

Reply:
The naming hinges on whether our field-bearing class extends what we presently call `MapExpression<Expression>`. If it does, then it seems reasonable to call it a document. If it does not, then it should probably be called a record (but this seems odd, because now nothing is called a document).

Reply:
How does extending or not extending `MapExpression<Expression>` removes the overloaded nature of the word "document"? Whether `DocumentExpression` is `MapExpression<Expression>` or not, "document" in `DocumentExpression` (what I called "MQL document" above) has still a different meaning than "BSON document" because item 2 [here](https://github.com/mongodb/mongo-java-driver/pull/1032#discussion_r1062674087) still holds.

Reply:
I believe we discussed and resolved this, that it will remain called Document?

Reply:
Yes, there is nothing left to do or discuss here.